### PR TITLE
[Feature] Support custom set kv buffer kernel in srt

### PR DESCRIPTION
**Pull request by DarkSharpness**
## Motivation

This PR is the integration of #8884 in Python code.

## Modifications

Integrate custom set_kv_buffer_kernel into `MHATokenToKVPool`.

## Benchmarking and Profiling

```
python3 -m sglang.launch_server --model meta-llama/Llama-3.1-70B-Instruct --load-format dummy --prefill-attention fa3 --decode-attention flashinfer --tp 8 --disable-radix
python3 -m sglang.bench_serving --backend sglang --dataset-name random --num-prompts 10 --random-input 4096 --random-output 1024 --random-range-ratio 1 --warmup-requests 1
```

Before:

```
============ Serving Benchmark Result ============
Backend:                                 sglang    
Traffic request rate:                    inf       
Max request concurrency:                 not set   
Successful requests:                     10        
Benchmark duration (s):                  13.57     
Total input tokens:                      40960     
Total generated tokens:                  10240     
Total generated tokens (retokenized):    10752     
Request throughput (req/s):              0.74      
Input token throughput (tok/s):          3018.83   
Output token throughput (tok/s):         754.71    
Total token throughput (tok/s):          3773.54   
Concurrency:                             9.99      
----------------End-to-End Latency----------------
Mean E2E Latency (ms):                   13550.59  
Median E2E Latency (ms):                 13551.41  
---------------Time to First Token----------------
Mean TTFT (ms):                          1219.83   
Median TTFT (ms):                        1320.46   
P99 TTFT (ms):                           1765.93   
---------------Inter-Token Latency----------------
Mean ITL (ms):                           12.05     
Median ITL (ms):                         11.52     
P95 ITL (ms):                            12.03     
P99 ITL (ms):                            12.24     
Max ITL (ms):                            1479.45   
==================================================
```

After:

```
============ Serving Benchmark Result ============
Backend:                                 sglang    
Traffic request rate:                    inf       
Max request concurrency:                 not set   
Successful requests:                     10        
Benchmark duration (s):                  13.36     
Total input tokens:                      40960     
Total generated tokens:                  10240     
Total generated tokens (retokenized):    10752     
Request throughput (req/s):              0.75      
Input token throughput (tok/s):          3067.01   
Output token throughput (tok/s):         766.75    
Total token throughput (tok/s):          3833.77   
Concurrency:                             9.99      
----------------End-to-End Latency----------------
Mean E2E Latency (ms):                   13339.98  
Median E2E Latency (ms):                 13341.00  
---------------Time to First Token----------------
Mean TTFT (ms):                          1200.47   
Median TTFT (ms):                        1302.12   
P99 TTFT (ms):                           1727.99   
---------------Inter-Token Latency----------------
Mean ITL (ms):                           11.87     
Median ITL (ms):                         11.35     
P95 ITL (ms):                            11.91     
P99 ITL (ms):                            12.34     
Max ITL (ms):                            1452.32   
==================================================
```

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

### CODE REVIEW

File: python/sglang/srt/mem_cache/memory_pool.py [hunk 1]
```diff
@@ -33,6 +33,7 @@
 import torch
 import triton
 import triton.language as tl
+from sgl_kernel import set_kv_buffer_kernel
```
Comment (zhyncs) [inline python/sglang/srt/mem_cache/memory_pool.py]: shall we only import sgl_kernel in cuda env?

Comment (DarkSharpness) [reply python/sglang/srt/mem_cache/memory_pool.py]: sure. fixed
